addons/binding/org.openhab.binding.bigassfan/src/main/java/org/openhab/binding/bigassfan/handler/BigAssFanHandler.java

```diff
@@ -0,0 +1,1061 @@
+    }
+
+    private PercentType convertLevelToPercent(String level) {
+        // Produces dimmer state from values 0-16
+        return new PercentType((int) Math.round(Integer.parseInt(level) * 6.25));
+    }
+
+    /*
+     * Send a command to the fan
+     */
+    private void sendCommand(String mac, String commandFragment) {
+        if (fanListener == null) {
+            logger.error("Unable to send message because fanListener is null!!!");
+            return;
+        }
+
+        StringBuilder sb = new StringBuilder();
+        sb.append("<").append(mac).append(commandFragment).append(">");
+        String message = sb.toString();
+        logger.debug("Sending message to fan at {}: {}", ipAddress, message);
+        fanListener.send(message);
+    }
+
+    private void updateChannel(String channelName, State state) {
+        Channel channel = thing.getChannel(channelName);
+        if (channel != null) {
+            updateState(channel.getUID(), state);
+        }
+    }
+
+    private String thingID() {
+        return thing.getUID().getId();
+    }
+
+    /*
+     * Manage the ONLINE/OFFLINE status of the thing
+     */
+    private void markOnline() {
+        if (!isOnline()) {
+            logger.debug("Changing status of {} from {}({}) to ONLINE", thingID(), getStatus(), getDetail());
+            updateStatus(ThingStatus.ONLINE);
+        }
+    }
+
+    private void markOffline() {
+        if (isOnline()) {
+            logger.debug("Changing status of {} from {}({}) to OFFLINE", thingID(), getStatus(), getDetail());
+            updateStatus(ThingStatus.OFFLINE);
+        }
+    }
+
+    private void markOfflineWithMessage(ThingStatusDetail statusDetail, String statusMessage) {
+        // If it's offline with no detail or if it's not offline, mark it offline with detailed status
+        if ((isOffline() && getDetail().equals(ThingStatusDetail.NONE)) || !isOffline()) {
+            logger.debug("Changing status of {} from {}({}) to OFFLINE({})", thingID(), getStatus(), getDetail(),
+                    statusDetail);
+            updateStatus(ThingStatus.OFFLINE, statusDetail, statusMessage);
+            return;
+        }
+    }
+
+    private boolean isOnline() {
+        return thing.getStatus().equals(ThingStatus.ONLINE);
+    }
+
+    private boolean isOffline() {
+        return thing.getStatus().equals(ThingStatus.OFFLINE);
+    }
+
+    private ThingStatus getStatus() {
+        return thing.getStatus();
+    }
+
+    private ThingStatusDetail getDetail() {
+        return thing.getStatusInfo().getStatusDetail();
+    }
+
+    /**
+     * The {@link FanListener} is responsible for sending and receiving messages to a fan.
+     *
+     * @author Mark Hilbush - Initial contribution
+     */
+    public class FanListener {
+        private final Logger logger = LoggerFactory.getLogger(FanListener.class);
+
+        // Our own thread pool for the long-running listener job
```

Comment:
> Previous guidance for long-running threads was to create my own thread rather than use the pool.

The guideline is indeed about scheduling jobs, which should be done through the existing thread pools. If I understand your code correctly, your background discovery requires a permanent thread on its own, which waits for messages. In such a case it indeed does not make sense to permanently block a thread from the pool, but it is ok to use your own.